pkg/varcontext/interpolation/funcs.go

```diff
@@ -147,3 +150,57 @@ func hilFuncAssert() ast.Function {
 		},
 	}
 }
+
+// hilFuncJsonMarshal marshals a value as JSON.
+func hilFuncJsonMarshal() ast.Function {
+	return ast.Function{
+		ArgTypes:   []ast.Type{ast.TypeAny},
+		ReturnType: ast.TypeString,
+		Callback: func(args []interface{}) (interface{}, error) {
+
+			unwrapped, err := hilToInterface(args[0])
+			if err != nil {
+				return nil, err
+			}
+			bytes, err := json.Marshal(unwrapped)
+			if err != nil {
+				return nil, fmt.Errorf("couldn't convert: %v to JSON %s", args[0], err)
+			}
+			return string(bytes), nil
+		},
+	}
+}
+
+func hilToInterface(arg interface{}) (interface{}, error) {
```

Comment:
@josephlewis42 and I talked. Since this and the other function are not exposed, and they've been tested at a higher level (see eval_test.go), there wouldn't be much value to the unit tests here.